### DISCUSSION TITLE
AST & libraries for AsciiDoc (compatible with mdast & remark to some extent)

### DISCUSSION THREAD

**Discussion by tats-u** [syntax-tree/unist#102] [Ideas] [answered]
AsciiDoc (https://asciidoc.org/ / https://asciidoctor.org/) is an HTML-compatible markup language similar to Markdown.
As of now, there is no pure-JS/TS-compatible library to convert AsciiDoc to HTML unlike Markdown, so much fewer systems support AsciiDoc than Markdown.
The reason I prefer Markdown is AsciiDoc has more built-in features than Markdown; AsciiDoc supports description lists, tables with colspan/rowspan, and admonitions without ugly raw HTML tags.
It's so disappointing Markdown, with fewer features, rules the document formats' world.
I wish the specification of an AST for AsciiDoc similar to mdast is officially specified and packages using the AsciiDoc AST like remark are officially developed.

**Comment by wooorm** [comment]
This isn’t very actionable? Not sure what you want to discuss!

For “official”, nothing done here is “official” to markdown, it follows existing specs, but the “official” part is those specs!

But: AsciiDoc is the best AsciiDoc out there. Markdown is the best markdown out there.

**Reply by tats-u** [reply]
You, unified, have designed the specifications of ASTs of not only Markdown but also HTML and *XML*.
AsciiDoc is compatible with HTML, like Markdown, so I asked you how much you're interested in the design for an AsciiDoc AST.
If not at all, I'll do by myself when I have time.

P.S. Can someone tell me why unified decided to design the XML AST (xast)? Aren't Markdown and HTML sufficient? Is it just because XML is similar to HTML?

**Reply by wooorm** [reply, marked as answer]
ah, yeah, I understand.

I am personally not interested in putting in the time to do this. But I can help advise you on how to do stuff if you want to do most of the work.

> P.S. Can someone tell me why unified decided to design the XML AST (xast)? Aren't Markdown and HTML sufficient? Is it just because XML is similar to HTML?

Why there are two ecosystems for markdown and HTML, is explained here: https://github.com/remarkjs/remark-rehype/tree/main#what-is-this.
There less of a difference between XML and HTML, but they’re still different!
